src/linker/implementation.py

```diff
@@ -0,0 +1,37 @@
+from pathlib import Path
+
+from linker.step import Step
+from linker.utilities.general_utils import load_yaml
+
+
+class Implementation:
+    def __init__(self, step: Step, directory: Path):
+        self.step = step
```

Comment:
make attributes private (if confirmed they aren't used externally after we refactor to run implementation.run)